httpx/_models.py

```diff
@@ -71,8 +71,10 @@ def __init__(self, url: URLTypes = "", params: QueryParamTypes = None) -> None:
                 # We don't want to normalize relative URLs, since doing so
                 # removes any leading `../` portion.
                 self._uri_reference = self._uri_reference.normalize()
-        else:
+        elif isinstance(url, URL):
             self._uri_reference = url._uri_reference
+        else:
+            raise TypeError("Invalid type for url.  Expected str or httpx.URL")
```

Comment:
(Nit) Consider showing the received type of `url`

```suggestion
            raise TypeError(f"Expected url to be an str or httpx.URL, got {type(url)}")
```

Reply:
@florimondmanca great suggestion, thanks 👍